Add Gustafson-Kessel clustering algorithm

tomas-remis opened the pull request:
## PR description
<!-- A short description of what is done in this PR. -->
Implemented Gustafson-Kessel clustering algorithm.
## Before 
<!-- Small example showing the functionality before this PR.
Needed only if you are changing existing source code (e.g. bug fix) -->

## After
<!-- Small example showing the functionality added/changed in this PR. 
Needed only if you change the source code. -->
- Added Gustafson-Kessel in genfis.jl
- Added Gustafson-Kessel unit test in test_genfis.jl
- Added LinearAlgebra and Statistics packages as project dependencies
## Related issues
<!--
List the issues related to this PR. E.g.
- #01
- #02

If you are closing some issues add "fixes" before the issue number, e.g.
- fixes #01
- #02
-->

## Checklist
<!-- Needed only if you change the source code.
You don't need to get everything done before opening the PR :) -->
- [x] Added Gusstafson-Kessel base algorithm
- [x] Added tests
- [x] Added docstring

## Other
<!-- Add here any other relevant information. -->

tomas-remis committed:
Added Gustafson-Kessel clustering to genfis.jl

tomas-remis committed:
Created tests for Gustafson-Kessel

Commit:
Added LinearAlgebra and Statistics library dependencies

Commit:
Implemented working unit test for Gustafson-Kessel

lucaferranti commented:
I think the last item on your todo list would better fit a separate follow-up PR

lucaferranti commented on Project.toml:
```diff
@@ -7,20 +7,24 @@ version = "0.1.3"
 Dictionaries = "0.3"
 DocStringExtensions = "0.9"
 LightXML = "0.9"
+LinearAlgebra = "1.11.0"
```
```suggestion
LinearAlgebra = "1.6"
```

since we want the library to work also on older julia versions (starting to 1.6) we want to set the lower bound to `1.6`, similar for `Statistics`

lucaferranti replied:
also I think you'll need `]add Random` and add a lower bound (also 1.6)

tomas-remis replied:
Sorry for the wait, I updated the project dependencies.

tomas-remis committed:
Update Project.toml

Added Random package
Set minimal version of Random, LinearAlgebra and Statistics to 1.6
